src/ReCaptcha.php

```diff
@@ -0,0 +1,344 @@
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+ * DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE
+ * FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+ * DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+ * SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+ * CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+ * OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+namespace Google\ReCaptcha;
+
+use Nyholm\Psr7\Factory\Psr17Factory;
+use Symfony\Component\HttpClient\Psr18Client;
+use Psr\Http\Client\ClientInterface as PsrClient;
+use Psr\Http\Message\StreamFactoryInterface as StreamFactory;
+use Psr\Http\Message\RequestFactoryInterface as RequestFactory;
+
+class ReCaptcha
+{
+    use HandlesConstraints;
+    use ParsesResponse;
+
+    /**
+     * Version of this client library.
+     *
+     * @const string
+     */
+    public const VERSION = 'php_2.0.0';
+
+    /**
+     * URL for reCAPTCHA `siteverify` API.
+     *
+     * @const string
+     */
+    public const SITE_VERIFY_URL = 'https://www.google.com/recaptcha/api/siteverify';
+
+    /**
+     * Default state of the constraints array
+     *
+     * @var array
+     */
+    public const CONSTRAINTS_ARRAY = [
+        'hostname' => null,
+        'apk_package_name' => null,
+        'action' => null,
+        'threshold' => null,
+        'challenge_ts' => null,
+    ];
+
+    /**
+     * The PSR18 Client
+     *
+     * @var \Psr\Http\Client\ClientInterface
+     */
+    protected $client;
+
+    /**
+     * Request Factory
+     *
+     * @var \Psr\Http\Message\RequestFactoryInterface
+     */
+    protected $request;
+
+    /**
+     * Stream Factory
+     *
+     * @var \Psr\Http\Message\StreamFactoryInterface
+     */
+    protected $stream;
+
+    /**
+     * Secret to authenticate the account with the reCAPTCHA servers
+     *
+     * @var string
+     */
+    protected $secret;
+
+    /**
+     * ReCaptcha constructor.
+     *
+     * @param  \Psr\Http\Client\ClientInterface $client
+     * @param  \Psr\Http\Message\RequestFactoryInterface $request
+     * @param  \Psr\Http\Message\StreamFactoryInterface $stream
+     */
+    public function __construct(PsrClient $client, RequestFactory $request, StreamFactory $stream)
+    {
+        $this->client = $client;
+        $this->request = $request;
+        $this->stream = $stream;
+    }
+
+    /**
+     * Returns the HTTP Client to use against the reCAPTCHA servers.
+     *
+     * @return \Psr\Http\Client\ClientInterface
+     */
+    public function getClient()
+    {
+        return $this->client;
+    }
+
+    /**
+     * Sets the HTTP Client to use with this.
+     *
+     * @param  mixed $client The object that this client will use for
```

Comment:
Both getter, property use the interface
For me you should replace mixed by the interface